projects/RabbitMQ.Client/client/impl/WireFormatting.cs

```diff
@@ -428,7 +428,7 @@ public static unsafe int WriteShortstr(Span<byte> span, string val)
             {
                 try
                 {
-                    int bytesWritten = Encoding.UTF8.GetBytes(chars, val.Length, bytes, maxLength);
+                    int bytesWritten = val.Length > 0 ? Encoding.UTF8.GetBytes(chars, val.Length, bytes, maxLength) : 0;
```

Comment:
What's the purpose of the val.Length check? GetBytes returns 0 for 0 length inputs?

Reply:
The `fixed (char* chars = val)` assigns a null pointer to "chars" if "val" is an empty string and GetBytes throws an exaption if "chars" is null. Hence the val.Length check to support empty strings.

Reply:
Sorry, correction (actually removed the val.Length to verify) if the provided span only has space for the length bytes (4) then `fixed (byte* bytes = &span.Slice(4).GetPinnableReference())` produces a null pointer in "bytes".
It is to satisfy the first test case in `TestWriteLongstr_BytesWritten`